feat: Kotlin 2.2.0 release (#4911)
JetBrains/kotlin-web-site: docs/topics/whatsnew/whatsnew22.md

* The parameters for experimental commonization modes are [deprecated in Kotlin 2.2.0](compatibility-guide-22.md#deprecate-commonization-parameters). 
  Clear the commonization cache to delete invalid compilation artifacts.
* The deprecated `konanVersion` property is now [removed from the `CInteropProcess` task](compatibility-guide-22.md#deprecate-konanversion-in-cinteropprocess). 
  Use `CInteropProcess.kotlinNativeVersion` instead.
* Usage of the deprecated `destinationDir` property will now [lead to an error](compatibility-guide-22.md#deprecate-destinationdir-in-cinteropprocess). 
  Use `CInteropProcess.destinationDirectory.set()` instead.

## Documentation updates

This release brings notable documentation changes, including the migration of Kotlin Multiplatform documentation to the [KMP portal](https://www.jetbrains.com/help/kotlin-multiplatform-dev/get-started.html). 

Additionally, we launched a documentation survey, created new pages and tutorials, and revamped existing ones. 

### Kotlin's documentation survey

> Have you used Kotlin's documentation recently? We're looking for genuine feedback to make it better.

Reviewer: ```suggestion
We're looking for genuine feedback to make the Kotlin docs better.
```

Author: True! Nice catch.
